pyee/_base.py

```diff
@@ -80,6 +80,10 @@ def _add_event_handler(self, event, k, v):
     def _emit_run(self, f, args, kwargs):
         f(*args, **kwargs)
 
+    def event_names(self):
+        # Returns an array listing the events
+        return list(self._events.keys())[1:]
```

Comment:
This slice is dropping the first key:

```
❯ python
Python 3.8.8 (default, Apr 13 2021, 19:58:26) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> xs = [1, 2, 3]
>>> xs[1:]
[2, 3]
>>> 
```

I don't think this is what we want. I think you were trying to clone the list:

```
>>> xs = [1, 2, 3]
>>> xs[:]
[1, 2, 3]
```
The motivation for doing this is so that `self._events` may mutate before you've completed iterating over its keys - the list gets that out of the way for us. Good thinking! But I think that's already handled by converting it to a list:

```
>>> xs is xs
True
>>> xs is xs[:]
False
>>> xs is list(xs)
False
```

In other words, try the following:

```py
def event_names(self):
    return list(self._events.keys())
```

Reply:
I wanted to drop the first key becuse in the first place always new_listener, and it's not a really listener, or am i wrong?

ok, l will do it

Reply:
What's the behavior in node.js? do they filter out internal events there? I think the easiest thing will be to copy that interface.